tests/compression_setup.rs

```diff
@@ -0,0 +1,102 @@
+mod testsupport;
+
+use samsa::prelude::{
+    protocol::{self, produce::request::Attributes},
+    BrokerConnection, Compression, Error, KafkaCode, TcpConnection,
+};
+use std::collections::HashMap;
+
+const CLIENT_ID: &str = "writing and reading using compression setup";
+const CORRELATION_ID: i32 = 1;
+const PARTITION_ID: i32 = 0;
+
+#[tokio::test]
+async fn writing_and_reading_using_compression_setup() -> Result<(), Box<Error>> {
```

Comment:
lets copy the same structure from the Read&Write 1M Messages with Compression test, so that it is avioding the low level request/response building.

Keep the compression but only produce and read 5 messages instead of 1M